### PR TITLE
Plugin testing

### PR DESCRIPTION
**Pull request by lizlouise1335**
Another plugin test and small plugin if-statement tweak

### CODE REVIEW

**Review by JGSweets — APPROVED** [review]

File: dataprofiler/tests/plugins/test_plugins.py [hunk 1]
```diff
@@ -2,22 +2,44 @@
...
+            test_get_dict = get_plugins("test")
             self.assertDictEqual({"mock_test": test_plugin}, test_get_dict)
+
+    @mock.patch("dataprofiler.plugins.__init__.importlib.util")
+    @mock.patch("dataprofiler.plugins.__init__.os.path.isdir")
+    @mock.patch("dataprofiler.plugins.__init__.os.listdir")
+    def test_load_plugin(self, mock_listdir, mock_isdir, mock_importlib_util):
+        mock_listdir.side_effect = (
+            lambda dir: ["__pycache__", "py"]
+            if dir.endswith("plugins")
```
Comment (JGSweets) [inline dataprofiler/tests/plugins/test_plugins.py:31]: `dir` -> `dirname`

File: dataprofiler/tests/plugins/test_plugins.py [hunk 2]
```diff
@@ -2,22 +2,44 @@
...
+            else ["stillnotrealpy", "a.json", None]
+        )
+        mock_isdir.return_value = True
+        mock_importlib_util.spec_from_file_location.return_value = None
+        load_plugins()
+        mock_importlib_util.spec_from_file_location.assert_not_called()
+
+        mock_listdir.side_effect = (
+            lambda dir: ["folder"] if dir.endswith("plugins") else ["file.py"]
```
Comment (JGSweets) [inline dataprofiler/tests/plugins/test_plugins.py:40]: `dir` -> `dirname`

**Review by JGSweets — CHANGES_REQUESTED** [review]

File: dataprofiler/tests/plugins/test_plugins.py (hunk 1 above)
Comment (taylorfturner) [inline dataprofiler/tests/plugins/test_plugins.py:31]: change `dir` var name to something 
- more descriptive 
- not a python specific / built-in variable name

File: dataprofiler/tests/plugins/test_plugins.py (hunk 2 above)
Comment (taylorfturner) [inline dataprofiler/tests/plugins/test_plugins.py:40]: same above

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by JGSweets — APPROVED** [review]

**Review by taylorfturner — APPROVED** [review]
LGTM

**Pull request merged**